File: php/features.php
<?php
declare(strict_types=1);

// IpIntelligenceApi2 SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/RatelimitFeature.php';
require_once __DIR__ . '/feature/RetryFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';
require_once __DIR__ . '/feature/TimeoutFeature.php';


class IpIntelligenceApi2Features
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IpIntelligenceApi2BaseFeature();
            case "ratelimit":
                return new IpIntelligenceApi2RatelimitFeature();
            case "retry":
                return new IpIntelligenceApi2RetryFeature();
            case "test":
                return new IpIntelligenceApi2TestFeature();
            case "timeout":
                return new IpIntelligenceApi2TimeoutFeature();
            default:
                return new IpIntelligenceApi2BaseFeature();
        }
    }

    /**
     * Does a generated feature class back this name? False for a name only
     * an options extend instance can supply (the station adopt path) - the
     * constructor uses this to skip make_feature for such names instead of
     * adding a stray BaseFeature.
     */
    public static function has_feature(string $name): bool
    {
        switch ($name) {
            case "base":
            case "ratelimit":
            case "retry":
            case "test":
            case "timeout":
                return true;
            default:
                return false;
        }
    }
}

File: rb/features.rb
# IpIntelligenceApi2 SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module IpIntelligenceApi2Features
  def self.make_feature(name)
    case name
    when "base"
      IpIntelligenceApi2BaseFeature.new
    when "ratelimit"
      IpIntelligenceApi2RatelimitFeature.new
    when "retry"
      IpIntelligenceApi2RetryFeature.new
    when "test"
      IpIntelligenceApi2TestFeature.new
    when "timeout"
      IpIntelligenceApi2TimeoutFeature.new
    else
      IpIntelligenceApi2BaseFeature.new
    end
  end
end
